scripts/samplesheet_generator.py

```diff
@@ -12,6 +12,9 @@
 
 DESC = """EPP used to create HiseqX samplesheets"""
 
+# Pre-compile regexes in global scope:
+idx_pat = re.compile("([ATCG]{4,})-?([ATCG]*)")
+tenx_pat = re.compile("SI-GA-[A-H][1-9][0-2]?")
```

Comment:
It's nice to have globals in FULL CAPS; it makes them easier to spot in the code.